src/NuGet.Core/NuGet.Configuration/Settings/ImmutableSettings.cs

```diff
@@ -0,0 +1,62 @@
+// All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace NuGet.Configuration
+{
+    // Represents a wrapper for an immutable settings instance.
+    // This means that any methods invoked on this instance that try to alter it will throw.
+    internal class ImmutableSettings : ISettings
+    {
+        private readonly ISettings _settings;
+
+        internal ImmutableSettings(ISettings settings)
+        {
+            _settings = settings ?? throw new ArgumentNullException(string.Format(CultureInfo.CurrentCulture, Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(settings)));
```

Comment:
1.  The argument order is wrong.  See [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception.-ctor?view=netframework-4.8#System_ArgumentNullException__ctor_System_String_System_String_).
2.  Simply remove the explicit message; you don't need it and the part about "empty" is wrong.  The exception's default message works just fine.

Reply:
You're right on 1).  My mistake.  My visual parser is buggy.